### ISSUE TITLE
grains: Confusing messages when **not** using BigInteger

### ISSUE THREAD

**Issue by NobbZ** [exercism/fsharp#402]
Currently the testsuite assumes and the stub encourages the use of `BigInteger` as the return type for both, `total` and `square`.

Since I tend to use the “smallest” datatype that suites the expected co-domain, I ignored the `BigInteger` part in the types and replaced them by `uint64`.

This lead to funny failing tests:

```plain
Failed   GrainsTest.Test square 1
Error Message:
 FsUnit.Xunit+MatchException : Exception of type 'FsUnit.Xunit+MatchException' was thrown.
Expected: Equals 1
Actual:   was 1
```

Is there a way to either:

* Improve the messages to mention the type?
* Make the testsuite accepting an `uint64` and `bigint` as equal when they are?
* Use `uint64` for the exercise, since it is big enough for this exercise and more performant than a `BigInteger` and less expensive on memory?

**Comment by ErikSchierboom** [comment]
Good catch.

> Improve the messages to mention the type?
We can't if we keep using FsUnit as they do some weird things when comparing types that should be able to be converted to each other.

> Make the testsuite accepting an uint64 and bigint as equal when they are?
We can't if we use FsUnit, but we can if we would encode the tests as standard xUnit assertions.

> Use uint64 for the exercise, since it is big enough for this exercise and more performant than a BigInteger and less expensive on memory?
I think this is by far the best option. I've added a comment to #378 so that we will implement this when we create the test generator.

**Comment by ErikSchierboom** [comment]
Closed by #406
